lib/ptz.js

```diff
@@ -190,16 +231,51 @@ Cam.prototype.getConfigurations = function(callback) {
 		if (!err) {
 			var configurations = {};
 			data[0]['getConfigurationsResponse'].forEach(function(configuration) {
-				configurations[configuration['PTZConfiguration'][0]['name']] = {
-					useCount: parseInt(configuration['PTZConfiguration'][0]['useCount'])
-					, nodeToken: configuration['PTZConfiguration'][0]['nodeToken'][0]
-					, defaultPTZSpeed: {
-						x: configuration['PTZConfiguration'][0]['defaultPTZSpeed'][0]['panTilt'][0].$.x
-						, y: configuration['PTZConfiguration'][0]['defaultPTZSpeed'][0]['panTilt'][0].$.y
-						, zoom: configuration['PTZConfiguration'][0]['defaultPTZSpeed'][0]['zoom'][0].$.x
+				var pTZConfiguration = configuration['PTZConfiguration'][0];
+				var name = pTZConfiguration['name'];
+				// some cameras have this as an array instead of a string
+				if (Array.isArray(name)) {
+					name = name[0];
+				}
+				configurations[name] = {};
+				configurations[name].useCount = parseInt(pTZConfiguration['useCount']);
+				if ('$' in pTZConfiguration) {
+					configurations[name].token = pTZConfiguration['$']['token'];
+				}
+				if ('nodeToken' in pTZConfiguration) {
+					configurations[name].nodeToken = pTZConfiguration['nodeToken'][0];
+				}
+				if ('defaultPTZSpeed' in pTZConfiguration) {
+					var defaultPTZSpeed = pTZConfiguration['defaultPTZSpeed'][0];
+					configurations[name].defaultPTZSpeed = {};
+					if ('panTilt' in defaultPTZSpeed) {
+						configurations[name].defaultPTZSpeed.x = defaultPTZSpeed['panTilt'][0].$.x;
+						configurations[name].defaultPTZSpeed.y = defaultPTZSpeed['panTilt'][0].$.y;
 					}
-					, defaultPTZTimeout: configuration['PTZConfiguration'][0]['defaultPTZTimeout'][0]
-				};
+					if ('zoom' in defaultPTZSpeed) {
+						configurations[name].defaultPTZSpeed.zoom = defaultPTZSpeed['zoom'][0].$.x;
+					}
+				}
+				if ('defaultPTZTimeout' in pTZConfiguration) {
+					configurations[name].defaultPTZTimeout = pTZConfiguration['defaultPTZTimeout'][0];
+				}
+				if ('panTiltLimits' in pTZConfiguration) {
+					configurations[name].panTiltLimits = {};
+					var panTiltLimits = pTZConfiguration['panTiltLimits'][0]['range'][0];
+					configurations[name].panTiltLimits.XRange = {};
+					configurations[name].panTiltLimits.XRange.min = panTiltLimits.XRange[0].min[0];
+					configurations[name].panTiltLimits.XRange.max = panTiltLimits.XRange[0].max[0];
+					configurations[name].panTiltLimits.YRange = {};
+					configurations[name].panTiltLimits.YRange.min = panTiltLimits.YRange[0].min[0];
+					configurations[name].panTiltLimits.YRange.max = panTiltLimits.YRange[0].max[0];
+				}
+				if ('zoomLimits' in pTZConfiguration) {
+					configurations[name].zoomLimits = {};
+					var zoomLimits = pTZConfiguration['zoomLimits'][0]['range'][0];
+					configurations[name].zoomLimits.XRange = {};
+					configurations[name].zoomLimits.XRange.min = zoomLimits.XRange[0].min[0];
+					configurations[name].zoomLimits.XRange.max = zoomLimits.XRange[0].max[0];
+				}
```

Comment:
What camera(s) have you tested this on just out of interest? I've got a few I can test on here too

Reply:
Hikvision, Pelco and Axis